grails-app/services/com/unifina/service/PermissionService.groovy

```diff
@@ -265,15 +268,92 @@ class PermissionService {
      * @return granted permission
      */
 	Permission systemGrant(Userish target, resource, Operation operation=Operation.READ) {
+		return systemGrant(target, resource, operation, null, null)
+	}
+
+	Permission systemGrant(Userish target, resource, Operation operation=Operation.READ, Subscription subscription, Date endsAt) {
 		target = target.resolveToUserish()
 		String userProp = getUserPropertyName(target)
 		String resourceProp = getResourcePropertyName(resource)
 
-		return new Permission(
+		Permission parentPermission = new Permission(
 			(resourceProp): resource,
 			(userProp): target,
 			operation: operation,
+			subscription: subscription,
+			endsAt: endsAt
 		).save(flush: true, failOnError: true)
+
+		// When a user is granted read access (subscriber) or write access (publisher) to a stream,
+		// we need to set the corresponding inbox stream permissions (see methods comments below).
+		if (userProp == "user" && resourceProp == "stream") {
```

Comment:
I agree with the comment. See similar discussion [here](https://github.com/streamr-dev/engine-and-editor/pull/583#discussion_r275511715), with a suggested solution but unfortunately it might not be feasible for this version of Grails.